patches/server/0981-Rewrite-the-Amethyst-logic-to-throw-both-BlockSpread.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jacob Litewski <jacob.litewski@gmail.com>
+Date: Sat, 20 May 2023 17:09:40 -0400
+Subject: [PATCH] Rewrite the Amethyst logic to throw both BlockSpreadEvent and
+ BlockGrowEvent
+
+Since Amethyst technically both spreads and grows, it should throw both events.
+This will alow plugin creators more control over how Amethyst blocks work.
+
+Signed-off-by: Jacob Litewski <jacob.litewski@gmail.com>
+
+diff --git a/src/main/java/net/minecraft/world/level/block/BuddingAmethystBlock.java b/src/main/java/net/minecraft/world/level/block/BuddingAmethystBlock.java
+index bedccb8717d08d5a60058445b04ddff149e7d36c..9b633f677ffc91baef3bfc09bf351e635c62085f 100644
+--- a/src/main/java/net/minecraft/world/level/block/BuddingAmethystBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/BuddingAmethystBlock.java
+@@ -26,27 +26,34 @@ public class BuddingAmethystBlock extends AmethystBlock {
+     @Override
+     public void randomTick(BlockState state, ServerLevel world, BlockPos pos, RandomSource random) {
+         if (random.nextInt(5) == 0) {
+-            Direction enumdirection = BuddingAmethystBlock.DIRECTIONS[random.nextInt(BuddingAmethystBlock.DIRECTIONS.length)];
+-            BlockPos blockposition1 = pos.relative(enumdirection);
+-            BlockState iblockdata1 = world.getBlockState(blockposition1);
++            // Paper start - Rewrite to throw both BlockSpreadEvent and BlockGrowEvent, and more readable variable names
++            Direction checkDirection = BuddingAmethystBlock.DIRECTIONS[random.nextInt(BuddingAmethystBlock.DIRECTIONS.length)];
++            BlockPos newPos = pos.relative(checkDirection);
++            BlockState checkData = world.getBlockState(newPos);
```

Comment:
Avoid changing the names of variables like this, as it creates unneeded diff

Reply:
Alright, I'll try to fix it right quick and update it

Reply:
Okay, pushed the new patch without the local variable renames